sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -878,6 +925,18 @@ def plot(self, *, ax=None, n_cols=3, line_kw=None, contour_kw=None):
         import matplotlib.pyplot as plt  # noqa
         from matplotlib.gridspec import GridSpecFromSubplotSpec  # noqa
 
+        if line_kw is not None and ice_lines_kw is not None:
+            warnings.warn(
+                "Both `line_kw` and `ice_lines_kw` are specified. `ice_lines_kw` "
+                "will take priority. Do not pass `line_kw` to silence this "
+                "warning."
+            )
+        if line_kw is not None and pd_line_kw is not None:
+            warnings.warn(
+                "Both `line_kw` and `pd_line_kw` are specified. `pd_line_kw` will "
+                "take priority. Do not pass `line_kw` to silence this warning."
+            )
```

Comment:
If we do not support `line_kw` + {`ice_line_kw`, `pd_line_kw`}, I would prefer to error here. 

Although it is a little nice to have `line_kw` set styles for both and then `ice_line_kw` and `pd_line_kw` can make adjustments. WDYT @glemaitre ?

Reply:
And in this case, we don't need any warning. Why not it seems to be as intuitive.